packages/flutter/test/widgets/image_test.dart

```diff
@@ -1568,6 +1568,75 @@ void main() {
       expect(imageCache.statusForKey(provider).live, false);
     });
   });
+
+  testWidgets('Rotated images have anti-aliasing.', (WidgetTester tester) async {
+    // Construct a PNG of 100x100 blue pixels as Image widget doesn't support raw RGBA.
+    Uint8List bytes;
+    await tester.runAsync(() async {
+      const int imageWidth = 100;
+      const int imageHeight = 100;
+      final Uint8List pixels = Uint8List.fromList(List<int>.generate(
+        imageWidth * imageHeight * 4,
+            (int i) => i % 4 < 2 ? 0x00 : 0xFF, // opaque blue
+      ));
+      final Completer<void> completer = Completer<void>();
+      ui.decodeImageFromPixels(
+        pixels, imageWidth, imageHeight, ui.PixelFormat.rgba8888,
+            (ui.Image image) async {
+          final ByteData byteData = await image.toByteData(
+              format: ui.ImageByteFormat.png);
+          bytes = byteData.buffer.asUint8List();
+          completer.complete();
+        },
+      );
+      await completer.future;
+    });
+
+    // Double check that our bytes are valid.
+    ui.Image uiImage;
+    await tester.runAsync(() async {
+      final Completer<void> completer = Completer<void>();
+      ui.decodeImageFromList(bytes, (ui.Image image) {
+        uiImage = image;
```

Comment:
This works because it supports raw image data.  Image.memory only supports encoder data.

Reply:
@dnfield : `Image.memory(bytes)` is using the PNG encoded `bytes`: https://github.com/flutter/flutter/pull/51656/files#diff-1f9334e95055c237f00b72bb642684d4R1588

Reply:
Also, a raw image data will trigger exception in `Image.memory`. In my case, there's no exception, just empty drawing result.

Reply:
Ahh ok.